### PR TITLE
Use sphinx instead of gitbook

### PR DESCRIPTION
**Pull request by chrisburr**
Self hosted GitBook has [been deprecated and unmaintained](https://github.com/GitbookIO/gitbook#%EF%B8%8F-deprecation-warning) for a long time. This PR moves from using GitBook to Sphinx which also allows:

* Importing Jupyter notebooks directly with https://nbsphinx.readthedocs.io/
* Better link checking (this PR fixes a lot of broken links
* The various bugs to be fixed (math in panels, nested panels, downloading files, ...)
* Easier plug-in writing for customising how code is added/tested
* Support for importing content from other places such as TWiki

Hosted example with these changes: https://chrisburr.me/starterkit-sphinx/

The source is mostly unchanged except for a few places extra whitespace is needed. The main difference is that inline math is now written with
```
`$ x $`
```
instead of `$$ x $$`.

It's easy to switch to another Sphinx theme if people have suggestions?

Once this is merged I'll move the Sphinx configuration into a Python package so we can keep the various GitBooks in sync more easily.

### CODE REVIEW

File: second-analysis-steps/building-decays-part2.md [hunk 1]
```diff
@@ -31,15 +32,17 @@ Note how we needed to use the `CombineParticles` from `GaudiConfUtils.Configurab
...
 
 {% callout "Why `ConfigurableGenerators`?" %}
+
 If we had tried to simply use `CombineParticles` inside our 
 `SimpleSelection`, we would have seen it fail with the following error
 
-```output
+```
 NameError: Could not instantiate Selection because input Configurable CombineParticles has default name. This is too unsafe to be allowed.
 ```
```
Comment (vlisovsk) [inline second-analysis-steps/building-decays-part2.md]: Can we avoid color-highlighting here? This line is treated as code currently.

Comment (chrisburr) [reply second-analysis-steps/building-decays-part2.md]: Fixed. I hadn't realised the default syntax highlighting is `python3` and not `none`: https://www.sphinx-doc.org/en/1.5/config.html#confval-highlight_language

File: first-analysis-steps/prerequisites.md [hunk 2]
```diff
@@ -85,7 +89,8 @@ If you can successfully execute all of the above steps, you are ready to go for
 the workshop!
 
 {% callout "Using Bash" %}
-The [Bash shell](http://en.wikipedia.org/wiki/Bash_(Unix_shell%29) will be used
+
+The [Bash shell](https://en.wikipedia.org/wiki/Bash_(Unix_shell%29) will be used
```
Comment (vlisovsk) [inline first-analysis-steps/prerequisites.md]: This link seems to be broken

Comment (chrisburr) [reply first-analysis-steps/prerequisites.md]: Well spotted!

As it's an unpleasant URL for markdown parsing I've replaced it with a shortened version:  http://cern.ch/go/gdJ9

File: first-analysis-steps/davinci-grid.md [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
-job](minimal-dv-job.html) and run it on the grid.
+job](minimal-dv-job) and run it on the grid.
 
 `ganga` is a program which you can use to interact with your grid
 jobs. 
 
-Before creating your first `ganga` job, open the script `ntuple-options.py`, obtained in the [previous lesson](minimal-dv-job.html), and comment out the lines taking the local input data: we will now use the data stored on grid.
+Before creating your first `ganga` job, open the script `ntuple-options.py`, obtained in the [previous lesson](minimal-dv-job), and comment out the lines taking the local input data: we will now use the data stored on grid.
 
 Also, you need to know the path to your data from Bookkeeping. In our case the path is `/MC/2016/Beam6500GeV-2016-MagDown-Nu1.6-25ns-Pythia8/Sim09c/Trig0x6138160F/Reco16/Turbo03/Stripping28r1NoPrescalingFlagged/27163002/ALLSTREAMS.DST`. Note, that here the event type number should be located at the end of the path, which is not the case if you browse the bookkeping by `Event type`.
```
Comment (vlisovsk) [inline first-analysis-steps/davinci-grid.md]: This path is also too long to fit in the page

File: first-analysis-steps/lhcb-dev.md [hunk 4]
```diff
@@ -37,6 +41,7 @@ cumbersome, for any serious development the usage of vanilla git is much more
 stable. Please consider using it if you can spare the compilation time.
```
Comment (vlisovsk) [inline first-analysis-steps/lhcb-dev.md]: It seems the link to the bottom of the page does not work.

Comment (vlisovsk) [reply first-analysis-steps/lhcb-dev.md]: Also, above (I cannot leave a comment there as you haven't edited that line): the link to gitlab is not clickable.

File: first-analysis-steps/lhcb-dev.md [hunk 5]
```diff
@@ -175,8 +185,10 @@ package and version:
 ```
 
 {% callout "What if `git` asks for my password?" %}
+
 Make sure you succesfully completed the instructions under [initial
 setup](#initial-setup).
```
Comment (vlisovsk) [inline first-analysis-steps/lhcb-dev.md]: This one doesn't seem to work either

Comment (chrisburr) [reply first-analysis-steps/lhcb-dev.md]: None of the new panels were being given ID of the heading which could be referenced. I've fixed it.

File: .travis.yml [hunk 6]
```diff
@@ -1,32 +1,22 @@
...
-  - gitbook install
-  - npm install svgexport -g
-# Using xvfb to Run Tests That Require a GUI
-# https://docs.travis-ci.com/user/gui-and-headless-browsers/#Using-xvfb-to-Run-Tests-That-Require-a-GUI
-before_script:
-  - "export DISPLAY=:99.0"
-  - sleep 3 # give xvfb some time to start
+  # FIXME: Inline math is broken in upstream recommonmark
+  - pip install git+https://github.com/chrisburr/recommonmark.git@patch-1
+  - pip install starterkit-ci
```
Comment (henryiii) [inline .travis.yml]: What is in starterkit-ci? Theme and some runners? Will it be here too (since it needs to also be maintained)?

**Comment by henryiii** [comment]
Can we still start callouts as hidden, etc?

File: .travis.yml (hunk 6 above)
Comment (chrisburr) [reply .travis.yml]: Yes and it's still present in some of the earlier commits. I've just made the repository for it and pushed it here: https://github.com/lhcb/starterkit-ci

**Comment by chrisburr** [comment]
> Can we still start callouts as hidden, etc?

Yes and they can all be toggled like before. The actual code for parsing the callouts is [here](https://github.com/lhcb/starterkit-ci/blob/master/src/starterkit_ci/sphinx_config/panels.py).

Example: https://chrisburr.me/starterkit-sphinx/second-analysis-steps/analysis-automation-snakemake.html#use-wildcards

**Comment by chrisburr** [comment]
To add, arbitrary deep nesting now works. The only changes are that `solution` must have an `endsolution` and there must be and empty line before and after the `{% callout_type "heading" %}` line (this could be avoided, but I didn't dig deep enough into `docutils` to figure out how).

**Comment by vlisovsk** [comment]
Is this ready to be merged?

**Comment by chrisburr** [comment]
I think so, I'll have some more changes which I'm developing on analysis-essentials but they can wait for another PR

**Comment by chrisburr** [comment]
Thanks everyone for reviewing and finding the issues! 🍰

**Pull request merged**